forem/forem
# E2E Test: Initial Adminisitrator Login Flow

## What type of PR is this? (check all applicable)

- [x] E2E test for critical flow in Forem

## Description

Introduces our first end to end test to the Forem codebase. I went with the first thing that happens if you create a new Forem instance. Logging in as the initial administrator.

## Related Tickets & Documents

#382, #11841 

## QA Instructions, Screenshots, Recordings

To test:

- Pull down this branch on your local, for example, via the [GitHub CLI](https://github.com/cli/cli), `gh co 11840`
- Run the following from the command line to install the latest dependencies which include Cypress, `yarn`.
- Ensure your local server is running, i.e. `http://localhost:3000` by running `bin/startup` from the command line.
- Run `yarn e2e` on another command line.
- Cypress will open

![image](https://user-images.githubusercontent.com/833231/101794667-1c2ba700-3ad5-11eb-91a2-5e740bd8cb32.png)

- Click on the Run 1 Integrated Spec link

![image](https://user-images.githubusercontent.com/833231/101793042-67dd5100-3ad3-11eb-96de-4971ab7baafe.png)

- The test runner will start

![image](https://user-images.githubusercontent.com/833231/101794759-32396780-3ad5-11eb-996d-e2ddbd4a07ab.png)

- The browser you have selected in the top right of the Cypress main window will run the tests successfully.

![image](https://user-images.githubusercontent.com/833231/101794816-43827400-3ad5-11eb-886b-2529b262e34c.png)

### UI accessibility concerns?

N/A

## Added tests?

      have not been included_

## Added to documentation?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Charile Rose from Shitt's Creek saying "Let's do it"](https://media.giphy.com/media/QvwCVnX9DWdlHCnix5/giphy-downsized-large.gif)
